STM32 Ethernet driver build failure with CONFIG_ASSERT=1

**Describe the bug**
STM32 Ethernet driver fails to build with CONFIG_ASSERT=1. This is introduced after the changeset under https://github.com/zephyrproject-rtos/zephyr/pull/25393 which added the following code, 
```
/* Now that the iface is setup, we are safe to enable IRQs. */
__ASSERT_NO_MSG(DEV_CFG(dev)->config_func() != NULL);
DEV_CFG(dev)->config_func();
```
The assertion is supposed to validate that the function call is valid while here we invoke the function and compare with its output. As the function returns `VOID`, we get a build error with assertion enabled. *Stupidity* on my part and 

**Expected behavior**
No build failure with Ethernet+Assert enabled. 

**Impact**
Unable to build / test Ethernet. **Regression.** 

**Screenshots or console output**
```
In file included from ../../../../../../include/kernel_includes.h:22,
                 from ../../../../../../include/kernel.h:17,
                 from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/ethernet/eth_stm32_hal.c:12:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/ethernet/eth_stm32_hal.c: In function 'eth_iface_init':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/ethernet/eth_stm32_hal_priv.h:48:3: error: void value not ignored as it ought to be
   48 |  ((const struct eth_stm32_hal_dev_cfg *)(dev)->config_info)
../../../../../../include/sys/__assert.h:86:9: note: in definition of macro '__ASSERT_NO_MSG'
   86 |   if (!(test)) {                                            \
      |         ^~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/ethernet/eth_stm32_hal.c:481:19: note: in expansion of macro 'DEV_CFG'
  481 |   __ASSERT_NO_MSG(DEV_CFG(dev)->config_func() != NULL);
      |                   ^~~~~~~
[48/165] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/scb.c.obj
```

--- a/drivers/ethernet/eth_stm32_hal.c
+++ b/drivers/ethernet/eth_stm32_hal.c
@@ -478,7 +478,7 @@ static void eth_iface_init(struct net_if *iface)
 		dev_data->iface = iface;
 
 		/* Now that the iface is setup, we are safe to enable IRQs. */
-		__ASSERT_NO_MSG(DEV_CFG(dev)->config_func() != NULL);
+		__ASSERT_NO_MSG(DEV_CFG(dev)->config_func != NULL);
 		DEV_CFG(dev)->config_func();
 	}
 

